front/lib/api/assistant/configuration.ts

```diff
@@ -69,56 +71,72 @@ export async function getAgentConfiguration(
   if (isGlobalAgentId(agentId)) {
     return await getGlobalAgent(auth, agentId, null);
   }
+
   const user = auth.user();
-  const agent =
-    preFetchedAgentConfiguration ??
-    (await AgentConfiguration.findOne({
-      where: {
-        sId: agentId,
-        workspaceId: owner.id,
-      },
-      order: [["version", "DESC"]],
-      include: [
-        {
-          model: AgentGenerationConfiguration,
-          as: "generationConfiguration",
-        },
-        {
-          model: AgentRetrievalConfiguration,
-          as: "retrievalConfiguration",
-        },
-        {
-          model: AgentDustAppRunConfiguration,
-          as: "dustAppRunConfiguration",
-        },
-        {
-          model: AgentDatabaseQueryConfiguration,
-          as: "databaseQueryConfiguration",
+
+  const [agent, userRelation] = await Promise.all([
```

Comment:
Do you think it's worth checking how to keep this to a single query in sequelize?
(since in pure SQL the way to keep it to a single query by adding an outer join is obvious)

Reply:
I think not worth right now as we have preFetch for all of this for the list of agents. We can def revisit in the future :+1: